libphal/libphal.H

```diff
@@ -278,6 +283,32 @@ bool isSbeVitalAttnActive(struct pdbg_target *proc);
  */
 bool hasControlTransitionedToHost();
 
+/**
+ * @brief Get unexpanded location code
+ *
+ * @details An api to get an unexpanded location code corresponding to
+ *          a given expanded location code. Unexpanded location codes
+ *          give the location of the FRU in the system.
+ *
+ * @param[in] locCode - Location code in expanded format.
+ *
+ * @return Location code as string which will be in un-expanded format or
+ *         empty optional on failure.
+ */
+std::optional<LocationCode> getUnexpandedLocCode(const LocationCode &locCode);
```

Comment:
this looks like a local function used in getFRUType if so do not define in header file or put in internal namepsace

Reply:
I added this in libphal.H as Ramesh mentioned there might be a usecase where we need unexpanded location code  in libphal https://github.com/open-power/ipl/pull/89#discussion_r1681218623